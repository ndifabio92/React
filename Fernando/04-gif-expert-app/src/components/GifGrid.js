// import React, { useState, useEffect } from 'react';
import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGitfs';
import { GifGridItem } from './GifGridItem';
import PropType from 'prop-types';
// import { getGifs } from '../helpers/getGifs';

export const GifGrid = ({ category }) => {

    // const [count, setCount] = useState(0);
    // const [ images, setImages ] = useState([]);
    const { data:images, loading } = useFetchGifs( category );
    // useEffect( () => {
    //     getGifs( category )
    //         .then( setImages );
    // }, [ category ] );

    return (
        <>
            <h3 className="animate__animated animate__bounce animate__delay-2s">{ category }</h3>
            { loading && <p className="animate__animated animate__bounce animate__flash">Loading</p> }
            <div className="card-grid">
                {
                    images.map( ( img ) => (
                        <GifGridItem 
                            key = { img.id } 
                            { ...img } // img = { img }
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropType.string.isRequired
}