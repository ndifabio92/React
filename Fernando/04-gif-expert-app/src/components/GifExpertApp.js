import React, { useState } from 'react'
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = () => {
    // const categories = ['one punch','samurai x','dragon ball'];

    const [ categories, setCategories ] = useState(['']);

    // const handleAdd = () => {
    //     setCategories( cats => [ ...cats, 'HunterXHunter']);
    // }

    return (
        <>
            <h2>GrifExpertApp</h2>
            <AddCategory setCategories = { setCategories }/>
            <hr/>
            {/* <button onClick={ () => handleAdd() }>Agregar</button> */}
            <ol>
                {/* { 
                    categories.map( category => {
                        return <li key={ category }> { category} </li>
                    })
                } */}
                {
                    categories.map( category => (
                        <GifGrid key = { category } category = { category } />
                    ))
                }
            </ol>
        </>
    )
}
