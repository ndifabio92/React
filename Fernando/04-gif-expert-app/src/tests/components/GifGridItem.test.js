import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

    const title = 'Un titulo';
    const url = 'https://localhost/algo.com';
    const wrapper = shallow( <GifGridItem title = { title } url = { url } /> );
    
    test('Debe mostrar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    });
    
    test('Debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('src') ).toBe( url );
    });

    test('Debe de tener animate__bounce animate__delay-2s', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect( className.includes('animate__bounce animate__delay-2s')).toBe( true );
    })
    
    
})
