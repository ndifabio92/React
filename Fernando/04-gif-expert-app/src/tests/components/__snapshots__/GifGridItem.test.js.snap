// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounce animate__delay-2s"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.com"
  />
  <p>
    Un titulo
  </p>
</div>
`;
